bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniVisitLocationController.java

```diff
@@ -32,4 +34,25 @@ public HashMap<String, String> getVisitLocationInfo(@PathVariable("loginLocation
         return visitLocation;
     }
 
+    @RequestMapping(method = RequestMethod.GET, value = "/facility/{locationUuid}")
+    @ResponseBody
+    public HashMap<String, String> getFacilityVisitLocationInfo(@PathVariable("locationUuid") String locationUuid ) {
+        Location location = Context.getLocationService().getLocationByUuid(locationUuid);
+        HashMap<String, String> facilityVisitLocation = new HashMap<>();
+        facilityVisitLocation.put("uuid", getParentVisitLocationUuid(location));
+        return facilityVisitLocation;
+    }
+
+    private String getParentVisitLocationUuid(Location location) {
```

Comment:
will this work for this situation, consider the following hierarchy
ABC Hospital Chain 
- Hospital 1 (Visit Location)
- Hospital 2 (Visit Location)

if the api is called with the location id of hospital 1, what will the api return? I think its missing checking whether the passed on location is a visit location on its own or not.

Reply:
In the given hierarchy, if ABC Hospital Chain is a visit location, it will return ABC Hospital Chain, else it will return Hospital 1 even if Hospital 1 is tagged with Visit Location or not.

If the parent doesn't have visit location tagged with it, then it will return the passed on location though it doesn't have visit location tag.

Reply:
not sure about that. Check the method getParentVisitLocationUuid() which you are calling with the identified location. The if clause checks with the isVisitLocation() function but you are passing only location.getParentLocation()

Reply:
modified the condition. Please check